src/txorphanage.cpp

```diff
@@ -319,3 +319,41 @@ std::vector<TxOrphanage::OrphanTxBase> TxOrphanage::GetOrphanTransactions() cons
     }
     return ret;
 }
+
+void TxOrphanage::SanityCheck() const
+{
+    // Check that cached m_total_announcements is correct
+    unsigned int counted_total_announcements{0};
+    // Check that m_total_orphan_size is correct
+    unsigned int counted_total_size{0};
+
+    // Check that cached PeerOrphanInfo::m_total_size is correct
+    std::map<NodeId, unsigned int> counted_size_per_peer;
+
+    for (const auto& [wtxid, orphan] : m_orphans) {
+        counted_total_announcements += orphan.announcers.size();
+        counted_total_size += orphan.GetSize();
+
+        for (const auto& peer : orphan.announcers) {
+            auto& count_peer_entry = counted_size_per_peer.try_emplace(peer).first->second;
+            count_peer_entry += orphan.GetSize();
+        }
+    }
+
+    Assume(counted_total_announcements == m_total_announcements);
```

Comment:
while we're here could also assert `m_total_announcements >=  m_total_orphan_size`

Reply:
That doesn't seem likely to be true? Do you mean `>= m_orphans.size()`?

Reply:
oops, yes

Reply:
(I added that instead)